### PR TITLE
login & registration modals implemented

### PR DESCRIPTION
**Pull request by xIllusivex**

### CODE REVIEW

**Review by ktan114 — APPROVED** [review]
Approved!

File: src/App.js [hunk 1]
```diff
@@ -78,15 +81,44 @@ const routes = [
...
+class Base extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      registerModal: false,
+      signInModal: false,
+    };
+  }
+  handleModalToggle = modalName => {
+    const modal =
```
Comment (ShawnStewart) [inline src/App.js]:
```suggestion
    const otherModal =
```
Maybe name this `otherModal` so it's a little more clear.

File: src/App.js [hunk 2]
```diff
@@ -78,15 +81,44 @@ const routes = [
...
+      modalName === 'registerModal' ? 'signInModal' : 'registerModal';
+    this.setState({ [modalName]: !this.state[modalName], [modal]: false });
```
Comment (ShawnStewart) [inline src/App.js]:
```suggestion
    this.setState((prevState) => { [modalName]: !prevState[modalName], [otherModal]: false });
```
`prevState` should be used to reference the previous state

File: src/components/LoginModal/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    this.state = {
+      showText: false,
+      email: '',
+      password: '',
+      error: false,
+      errorPhrase: 'Something went wrong.',
+    };
+  }
+  handleToggle = e => {
+    this.setState({ [e.target.name]: !this.state[e.target.name] });
```
Comment (ShawnStewart) [inline src/components/LoginModal/index.js:18]:
```suggestion
    this.setState((prevState) => { [e.target.name]: !prevState[e.target.name] });
```

**Review by ShawnStewart — APPROVED** [review]
I left some nitpick comments. Otherwise LGTM 👍

File: src/App.js (hunk 1 above)
Comment (xIllusivex) [reply src/App.js]: thanks for the suggestion

File: src/components/RegisterModal/index.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      password2: '',
+      error: false,
+    };
+  }
+  handleToggle = e => {
+    this.setState({ [e.target.name]: !this.state[e.target.name] });
```
Comment (ShawnStewart) [inline src/components/RegisterModal/index.js]:
```suggestion
    this.setState((prevState) => { [e.target.name]: !prevState[e.target.name] });
```

File: src/App.js (hunk 2 above)
Comment (xIllusivex) [reply src/App.js]: yo, nice i didnt know we could do that

**Pull request merged**